src/explore-education-statistics-common/src/modules/table-tool/components/TableHeadersReorderableList.tsx

```diff
@@ -0,0 +1,299 @@
+        return [...prevIndices, index];
+      }
+
+      const nextIndices = [...prevIndices];
+      nextIndices.splice(indexPosition, 1);
+
+      return nextIndices;
+    });
+  };
+
+  const performAction = (
+    event:
+      | React.MouseEvent<HTMLDivElement>
+      | React.KeyboardEvent<HTMLDivElement>,
+    index: number,
+  ) => {
+    if (isGroupKeyUsed(event)) {
+      toggleSelectionInGroup(index);
+      return;
+    }
+
+    toggleSelection(index);
+  };
+
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLDivElement>,
+    snapshot: DraggableStateSnapshot,
+    index: number,
+  ) => {
+    if (
+      event.defaultPrevented ||
+      snapshot.isDragging ||
+      event.key !== 'Enter'
+    ) {
+      return;
+    }
+
+    event.preventDefault();
+    performAction(event, index);
+  };
+
+  const handleClick = (
+    event: React.MouseEvent<HTMLDivElement>,
+    index: number,
+  ) => {
+    if (event.defaultPrevented || event.button !== primaryButton) {
+      return;
+    }
+
+    event.preventDefault();
+    performAction(event, index);
+  };
+
+  const handleTouchEnd = (
+    event: React.TouchEvent<HTMLDivElement>,
+    index: number,
+  ) => {
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    event.preventDefault();
+    toggleSelectionInGroup(index);
+  };
+
+  return (
+    <Field name={name}>
+      {({ field, form }: FieldProps) => {
+        const list: Filter[] = field.value;
+
+        return (
+          <FormFieldset legend={legend} legendSize="s" id={id}>
+            <div className={styles.focusContainer} tabIndex={-1}>
+              <DragDropContext
+                onDragStart={start => {
+                  if (!selectedIndices.includes(start.source.index)) {
+                    setSelectedIndices([]);
+                  }
+
+                  setDraggingIndex(start.source.index);
+                }}
+                onDragEnd={result => {
+                  if (result.destination?.index == null) {
+                    return;
+                  }
+
+                  const destinationIndex = result.destination.index;
+
+                  const selected = selectedIndices.length
+                    ? selectedIndices
+                    : [result.source.index];
+
+                  const nextValue = reorderMultiple({
+                    list,
+                    destinationIndex,
+                    selectedIndices: selected,
+                  });
+
+                  setDraggingIndex(undefined);
+
+                  const oldOptions = selected.map(index => list[index]);
+
+                  setSelectedIndices(
+                    nextValue.reduce<number[]>((acc, option, index) => {
+                      if (oldOptions.includes(option)) {
+                        acc.push(index);
+                      }
+
+                      return acc;
+                    }, []),
+                  );
+
+                  form.setFieldValue(name as string, nextValue);
```

Comment:
Don't think we need to cast `name` here (but could be wrong). Not getting any Typescript errors about it in my editor